lib/_stream_readable.js

```diff
@@ -919,6 +920,19 @@ Readable.prototype.wrap = function(stream) {
   return this;
 };
 
+var warningEmitted = false;
+Readable.prototype[Symbol.asyncIterator] = function() {
+  if (!warningEmitted) {
+    process.emitWarning(
```

Comment:
There's an `emitExperimentalWarning` somewhere in internal utils. We should start using it since it avoids needing to track `warningEmitted`. If it needs more features, we should extend it as needed.